src/Compilers/CSharp/Portable/Symbols/TypeWithAnnotations.cs

```diff
@@ -18,82 +18,23 @@ namespace Microsoft.CodeAnalysis.CSharp.Symbols
     [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
     internal readonly struct TypeWithAnnotations : IFormattable
     {
-        /// <summary>
-        /// A builder for lazy instances of TypeWithAnnotations.
-        /// </summary>
         [DebuggerDisplay("{GetDebuggerDisplay(), nq}")]
-        internal struct Builder
+        internal sealed class Boxed
         {
-            private TypeSymbol _defaultType;
-            private int _nullableAnnotation;
-            private Extensions _extensions;
-
-            /// <summary>
-            /// The underlying type, unless overridden by _extensions.
-            /// </summary>
-            internal TypeSymbol DefaultType => Volatile.Read(ref _defaultType);
-
-            /// <summary>
-            /// True if the fields of the builder are unset.
-            /// </summary>
-            internal bool IsDefault => Volatile.Read(ref _extensions) == null;
-
-            /// <summary>
-            /// Set the fields of the builder.
-            /// </summary>
-            /// <remarks>
-            /// This method guarantees: fields will be set once; exactly one caller is
-            /// returned true; and IsDefault will return true until all fields are initialized.
-            /// This method does not guarantee that all fields will be set by the same
-            /// caller. Instead, the expectation is that all callers will attempt to initialize
-            /// the builder with equivalent TypeWithAnnotations instances where
-            /// different fields of the builder may be assigned from different instances.
-            /// </remarks>
-            internal bool InterlockedInitialize(TypeWithAnnotations type)
-            {
-                if (!IsDefault)
-                {
-                    return false;
-                }
-
-                Interlocked.CompareExchange(ref _nullableAnnotation, (int)type.NullableAnnotation, 0);
-                Interlocked.CompareExchange(ref _defaultType, type.DefaultType, null);
-
-                // Because _extensions always gets a non-null value when the struct is initialized, we use it
-                // as a flag to signal that the initialization is complete. This means _extensions should be
-                // initialized last.
-                bool wasFirst = Interlocked.CompareExchange(ref _extensions, type._extensions ?? Extensions.Default, null) is null;
-
-                Debug.Assert(_extensions == (type._extensions ?? Extensions.Default));
-                Debug.Assert(_nullableAnnotation == (int)type.NullableAnnotation);
-                Debug.Assert(_defaultType.Equals(type.DefaultType));
-                return wasFirst;
-            }
+            internal readonly TypeWithAnnotations Value;
```

Comment:
Would it be useful to have an implicit conversion from Boxed to the underlying value?

Reply:
The reason I didn't was to make dereferences of `Boxed` be explicit in code. So we have usages of `_lazyType.Value` and `_lazyType?.Value`, but not implicit `TypeWithAnnotations x = _lazyType;`.

---
In reply to: [280908162](https://github.com/dotnet/roslyn/pull/35459#discussion_r280908162) [](ancestors = 280908162)